sql/Pg-upgrade2/file_object_type_as_enum.sql

```diff
@@ -50,3 +52,21 @@ UPDATE files SET object_type_new = object_type::file_object_types;
 ALTER TABLE files ALTER  COLUMN object_type_new SET NOT NULL;
 ALTER TABLE files DROP   COLUMN object_type;
 ALTER TABLE files RENAME COLUMN object_type_new TO object_type;
+
+-- constraint aus file_storage_purchase_quotation_intake wiederherstellen:
+ALTER TABLE files
+  ADD  CONSTRAINT valid_type CHECK (
+             (object_type = 'credit_note'                 ) OR (object_type = 'invoice'                   )
+          OR (object_type = 'sales_order'                 ) OR (object_type = 'sales_order_intake'        )
+          OR (object_type = 'sales_quotation'             ) OR (object_type = 'sales_delivery_order'      )
+          OR (object_type = 'request_quotation'           ) OR (object_type = 'purchase_quotation_intake' ) OR (object_type = 'purchase_order'          )
+          OR (object_type = 'purchase_delivery_order'     ) OR (object_type = 'purchase_invoice'          )
+          OR (object_type = 'vendor'                      ) OR (object_type = 'customer'                  ) OR (object_type = 'part'                    )
+          OR (object_type = 'gl_transaction'              ) OR (object_type = 'dunning'                   ) OR (object_type = 'dunning1'                )
+          OR (object_type = 'dunning2'                    ) OR (object_type = 'dunning3'                  ) OR (object_type = 'dunning_orig_invoice'    )
+          OR (object_type = 'dunning_invoice'             ) OR (object_type = 'draft'                     ) OR (object_type = 'statement'               )
+          OR (object_type = 'shop_image'                  ) OR (object_type = 'letter'                    ) OR (object_type = 'project'                 )
+          OR (object_type = 'invoice_for_advance_payment' ) OR (object_type = 'final_invoice'             ) OR (object_type = 'supplier_delivery_order' )
+          OR (object_type = 'sales_reclamation'           ) OR (object_type = 'purchase_reclamation'      ) OR (object_type = 'rma_delivery_order'      )
+  );
+
```

Comment:
die constraints wiederherstellen ist falsch. Genau deshalb gibt es ja das Upgrade-Skript, damit eben nicht immer alle Typen in eine valid_types-Contraint geschrieben werden müssen. So ist das Feld eben vom Typ file_object_types, der alle object types als ENUM aufgezählt hat. Diesen kann man dann bei Bedarf erweitern.